flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/BufferWritingResultPartition.java

```diff
@@ -111,6 +111,17 @@ public int getNumberOfQueuedBuffers() {
         return totalBuffers;
     }
 
+    @Override
+    public long getSizeOfQueuedBuffers() {
+        long totalNumberOfBytes = 0;
+
+        for (ResultSubpartition subpartition : subpartitions) {
+            totalNumberOfBytes += Math.max(0, subpartition.getTotalNumberOfBytes());
```

Comment:
I just notice that the access to getTotalNumberOfBytes is unsynchronized here.

Reply:
Rename `getSizeOfQueuedBuffers` to `getSizeOfQueuedBuffersUnsafe`?